Dockerfile

```diff
@@ -18,162 +19,284 @@
+#------------------------------------------------------------------------
+# Build components
+FROM ubuntu:focal AS build
 
-ARG ORT_VERSION="DOCKER-SNAPSHOT"
+# http://bugs.python.org/issue19846
+ENV LANG C.UTF-8
+
+# Set shell for bash and default pipefail
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
-# Apk install commands.
-RUN apk add --no-cache \
-        # Required for Node.js to build the reporter-web-app.
-        libstdc++ \
-        # Required to allow to download via a proxy with a self-signed certificate.
+#------------------------------------------------------------------------
+# Basic intro
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y --no-install-recommends \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        bash \
         ca-certificates \
-        coreutils \
-        openssl
+        curl \
+        gnupg
 
-COPY . /usr/local/src/ort
+#------------------------------------------------------------------------
+# Add git ppa for latest git
+RUN echo "deb http://ppa.launchpad.net/git-core/ppa/ubuntu focal main" > /etc/apt/sources.list.d/git-core-ubuntu-ppa-focal.list
+ENV KEYURL=https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xa1715d88e1df1f24
+RUN curl -ksS "$KEYURL" | gpg --dearmor | tee "/etc/apt/trusted.gpg.d/git-core_ubuntu_ppa.gpg"  > /dev/null
+
+#------------------------------------------------------------------------
+# Ubuntu build toolchain
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        build-essential \
+        dirmngr \
+	    dpkg-dev \
+        git \
+	    libbluetooth-dev \
+	    libbz2-dev \
+	    libc6-dev \
+	    libexpat1-dev \
+	    libffi-dev \
+        libgmp-dev \
+	    libgdbm-dev \
+	    liblzma-dev \
+        libmpdec-dev \
+	    libncursesw5-dev \
+	    libreadline-dev \
+	    libsqlite3-dev \
+	    libssl-dev \
+	    make \
+        netbase \
+        openjdk-11-jdk \
+	    tk-dev \
+        tzdata \
+        unzip \
+	    uuid-dev \
+	    xz-utils \
+	    zlib1g-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy the necessary bash resource to have paths
+# for different languages
+COPY docker/bash_bootstrap /etc/ort/bash_bootstrap
+RUN mkdir -p /etc/ort/bash_modules
+
+#------------------------------------------------------------------------
+# Python using pyenv
+ARG CONAN_VERSION=1.38.0
+ARG PYTHON_VERSION=3.8.11
+ARG SCANCODE_VERSION=21.8.4
+ARG VIRTUALENV_VERSION=20.2.2
+ARG PIP_VERSION=21.2.4
+
+# Python pyenv
+ENV PYENV_ROOT=/opt/python
+RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+ENV PATH=/opt/python/bin:$PATH
+RUN pyenv install "${PYTHON_VERSION}"
+RUN pyenv global "${PYTHON_VERSION}"
+ENV PATH=/opt/python/shims:$PATH
+RUN pip install -U pip=="${PIP_VERSION}" \
+    conan=="${CONAN_VERSION}" \
+    Mercurial \
+    scancode-toolkit[full]=="${SCANCODE_VERSION}" \
```

Comment:
they recommend a different way of installation (https://github.com/nexB/scancode-toolkit#installation), maybe use the code from https://github.com/oss-review-toolkit/ort/pull/4425, that matches the recommended way of installation.

Reply:
Fixed